staging/src/k8s.io/apiserver/pkg/server/mux/pathrecorder.go

```diff
@@ -74,12 +82,26 @@ func (m *PathRecorderMux) trackCallers(path string) {
 // refreshMuxLocked creates a new mux and must be called while locked.  Otherwise the view of handlers may
 // not be consistent
 func (m *PathRecorderMux) refreshMuxLocked() {
-	mux := http.NewServeMux()
+	newMux := mux.NewRouter().StrictSlash(true)
```

Comment:
> freshly assuming everything doing discovery is redirect-aware (and adding hops to discovery in general) seems bad

We should never have added those paths on top of routers that didn't deal with it.  I'd be willing to bit the bullet on it for the overall simplification.  The next best alternative may actually be writing our own mux, but to do that seems worse than our disease.